java/arcs/core/storage/handle/HandleManager.kt

```diff
@@ -44,6 +48,21 @@ interface ActivationFactoryFactory {
     fun <T : Referencable> setFactory(): CollectionActivationFactory<T>
 }
 
+/** Convenience class for dealing with a map of [Store] instances of any type */
+class Stores {
+    private val storesMutex = Mutex()
+    private val stores by guardedBy(storesMutex, mutableMapOf<StorageKey, Store<*, *, *>>())
+
+    @Suppress("UNCHECKED_CAST")
+    suspend fun <Data : CrdtData, Op : CrdtOperationAtTime, T> get(
+        storeOptions: StoreOptions<Data, Op, T>
+    ) = storesMutex.withLock {
+        stores.getOrPut(storeOptions.storageKey) {
+            Store(storeOptions)
+        } as Store<Data, Op, T>
+    }
+}
```

Comment:
This will get GC'd when the arc host goes away, right?  Do we need more explicit releasing-mechanisms here?

Reply:
Good question... the store might hold an active store, I wonder if we need to "close" the active store to make sure all callbacks are removed?

Reply:
I added a close() method to Handle recently but it just removes callbacks.  We could have a HandleManager.close() method that ArcHost uses when it shuts down an Arc, which closes all handles, stores, and cleans the cache, but we can do that in a follow on PR, so that we can unblock g3 related work.

Reply:
Sounds good!